azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -461,11 +461,11 @@ func resourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) er
 	if err := d.Set("consistency_policy", flattenAzureRmCosmosDBAccountConsistencyPolicy(resp.ConsistencyPolicy)); err != nil {
 		return fmt.Errorf("Error setting CosmosDB Account %q `consistency_policy` (Resource Group %q): %+v", name, resourceGroup, err)
 	}
-	if err := d.Set("failover_policy", flattenAzureRmCosmosDBAccountFailoverPolicy(resp.FailoverPolicies)); err != nil {
-		return fmt.Errorf("Error setting CosmosDB Account %q `failover_policy` (Resource Group %q): %+v", name, resourceGroup, err)
-	}
-	if err := d.Set("geo_location", flattenAndSetAzureRmCosmosDBAccountGeoLocations(d, resp)); err != nil {
-		return fmt.Errorf("Error setting CosmosDB Account %q `geo_location` (Resource Group %q): %+v", name, resourceGroup, err)
+	if _, ok := d.GetOk("failover_policy"); ok {
+		d.Set("failover_policy", flattenAzureRmCosmosDBAccountFailoverPolicy(resp.FailoverPolicies))
+	} else {
+		//if failover policy isn't default to using geo_location
+		d.Set("geo_location", flattenAzureRmCosmosDBAccountGeoLocations(d, resp))
 	}
```

Comment:
I'll push a commit to raise the errors if there's a failure here - this otherwise LGTM 👍